dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/AbstractK8sTaskExecutor.java

```diff
@@ -36,7 +38,9 @@ public abstract class AbstractK8sTaskExecutor {
     protected AbstractK8sTaskExecutor(TaskExecutionContext taskRequest) {
         this.taskRequest = taskRequest;
         this.k8sUtils = new K8sUtils();
-        this.yaml = new Yaml();
+        this.yaml = new Yaml(new ClassFilterConstructor(new Class[]{
+                List.class
+        }));
```

Comment:
Please also check this, nested non-primitive types should be added too

Reply:
![image](https://github.com/apache/dolphinscheduler/assets/34905992/06a6d15c-1857-492c-8837-4ee886a868fd)
According to docs, there is only List of Strings here.